File: templates/go-service/service/httphandlers/helloAll.go
package httphandlers

import (
	"fmt"
	"net/http"
	"strings"

	"github.com/gin-gonic/gin"
	"registry.com/user/servicename/context"
)

// CreateHelloAllHandler create helloPubSub handler
func CreateHelloAllHandler(context *context.Context) (handler func(c *gin.Context)) {
	return func(c *gin.Context) {
		greeting := "Hello, everyone..."
		context.InitLocalCache("names", []string{})
		names, _ := context.LocalCache.GetStringArray("names")
		if len(names) > 0 {
			greeting = fmt.Sprintf("Hello %s, and everyone...", strings.Join(names, ", "))
		}
		c.String(http.StatusOK, greeting)
	}
}

File: templates/nodejs-service/service/httphandlers/helloAll.js

function createHelloAllHandler(context) {
    return function (req, res) {
        context.initLocalCache("names", []);
        names = context.getLocalCache("names");
        const greeting = names.length == 0 ? "Hello, everyone..." : "Hello, " + names.join(", ") + " and everyone";
        res.send(greeting);
    }
}

module.exports = { createHelloAllHandler };